### ISSUE TITLE
Symbol encoding changes after defining a method with same name

### ISSUE THREAD

**Issue by fevin86** [jruby/jruby#4547]
### Environment

Provide at least:
* JRuby-9.1.8.0
* OSX, Fedora...

### Expected Behavior

```
a = :something
a.encoding # #<Encoding:US-ASCII>
def something; end
a.encoding # should be #<Encoding:US-ASCII>
```

### Actual Behavior

```
a = :something
a.encoding # #<Encoding:US-ASCII>
def something; end
a.encoding # it is now #<Encoding:ASCII-8BIT>
```

**Comment by kirillrdy** [comment]
Can I also add some comments about reproducing @fevin86 bug
 you need to evaluate the return of def, eg puts it
```ruby
a = :something
a.encoding # #<Encoding:US-ASCII>
puts def something; end
a.encoding # should be #<Encoding:US-ASCII>
```
the example he posted works if you paste it into irb, because irb evals return value of each line

**Comment by MSNexploder** [comment]
Should be the same issue as #4564 (fixed in 01cd84d).

**Comment by fevin86** [comment]
@MSNexploder you are right. Will close this
